### PR TITLE
Update deny config

### PR DESCRIPTION
**Pull request by tottoto**
## Motivation

Reflects the current state to `cargo-deny` config to fix `cargo-deny` test.

## Solution

Updates it. Refs #374.

### CODE REVIEW

**Review by Nehliin — APPROVED** [review]
Thanks!

**Comment by Nehliin** [comment]
Do you mind trying to get the CI api check fix in here as well? otherwise I can do it a bit later but you'll have to rebase afterwards

File: deny.toml [hunk 1]
```diff
@@ -36,6 +31,11 @@ license-files = [
 [bans]
 multiple-versions = "deny"
 highlight = "all"
+skip = [
+  # tracing-subscriber depends on multiple versions of regex-syntax.
+  # This is not actualy depended by tower-http as this is caused by examples.
```
Comment (jplatte) [inline deny.toml:36]:
```suggestion
  # This is not actually depended by tower-http as this is caused by examples.
```

Comment (tottoto) [reply deny.toml:36]: Thanks!

**Comment by tottoto** [comment]
@Nehliin Sure. I'll take a look it.

**Comment by Nehliin** [comment]
No I don't think we should remove it but it doesn't need to block this pr for now

**Pull request merged**